Description: allow named_t to read /proc/sys/vm/overcommit_memory
Author: Russell Coker <russell@coker.com.au>
Last-Update: 2014-01-12

Index: refpolicy/policy/modules/contrib/bind.te
===================================================================
--- refpolicy.orig/policy/modules/contrib/bind.te
+++ refpolicy/policy/modules/contrib/bind.te
@@ -113,6 +113,7 @@ read_files_pattern(named_t, named_zone_t
 read_lnk_files_pattern(named_t, named_zone_t, named_zone_t)
 
 kernel_read_kernel_sysctls(named_t)
+kernel_read_vm_overcommit_sysctl(named_t)
 kernel_read_system_state(named_t)
 kernel_read_network_state(named_t)
 
Index: refpolicy/policy/modules/contrib/devicekit.te
===================================================================
--- refpolicy.orig/policy/modules/contrib/devicekit.te
+++ refpolicy/policy/modules/contrib/devicekit.te
@@ -89,6 +89,7 @@ kernel_read_fs_sysctls(devicekit_disk_t)
 kernel_read_network_state(devicekit_disk_t)
 kernel_read_software_raid_state(devicekit_disk_t)
 kernel_read_system_state(devicekit_disk_t)
+kernel_read_vm_overcommit_sysctl(devicekit_disk_t)
 kernel_read_vm_sysctls(devicekit_disk_t)
 kernel_request_load_module(devicekit_disk_t)
 kernel_setsched(devicekit_disk_t)
Index: refpolicy/policy/modules/contrib/networkmanager.te
===================================================================
--- refpolicy.orig/policy/modules/contrib/networkmanager.te
+++ refpolicy/policy/modules/contrib/networkmanager.te
@@ -92,6 +92,7 @@ kernel_read_crypto_sysctls(NetworkManage
 kernel_read_system_state(NetworkManager_t)
 kernel_read_network_state(NetworkManager_t)
 kernel_read_kernel_sysctls(NetworkManager_t)
+kernel_read_vm_overcommit_sysctl(NetworkManager_t)
 kernel_request_load_module(NetworkManager_t)
 kernel_read_debugfs(NetworkManager_t)
 kernel_rw_net_sysctls(NetworkManager_t)
Index: refpolicy/policy/modules/contrib/virt.te
===================================================================
--- refpolicy.orig/policy/modules/contrib/virt.te
+++ refpolicy/policy/modules/contrib/virt.te
@@ -562,6 +562,7 @@ kernel_read_system_state(virtd_t)
 kernel_read_network_state(virtd_t)
 kernel_rw_net_sysctls(virtd_t)
 kernel_read_kernel_sysctls(virtd_t)
+kernel_read_vm_overcommit_sysctl(virtd_t)
 kernel_request_load_module(virtd_t)
 kernel_search_debugfs(virtd_t)
 kernel_setsched(virtd_t)
Index: refpolicy/policy/modules/kernel/kernel.if
===================================================================
--- refpolicy.orig/policy/modules/kernel/kernel.if
+++ refpolicy/policy/modules/kernel/kernel.if
@@ -3323,3 +3323,43 @@ interface(`kernel_unconfined',`
 	typeattribute $1 kern_unconfined;
 	kernel_load_module($1)
 ')
+
+########################################
+## <summary>
+##      Read virtual memory overcommit sysctl.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+## <rolecap/>
+#
+interface(`kernel_read_vm_overcommit_sysctl',`
+        gen_require(`
+                type sysctl_vm_overcommit_t;
+        ')
+
+        kernel_search_vm_sysctl($1)
+        allow $1 sysctl_vm_overcommit_t:file read_file_perms;
+')
+
+########################################
+## <summary>
+##      Read and write virtual memory overcommit sysctl.
+## </summary>
+## <param name="domain">
+##      <summary>
+##      Domain allowed access.
+##      </summary>
+## </param>
+## <rolecap/>
+#
+interface(`kernel_rw_vm_overcommit_sysctl',`
+        gen_require(`
+                type sysctl_vm_overcommit_t;
+        ')
+
+        kernel_search_vm_sysctl($1)
+        allow $1 sysctl_vm_overcommit_t:file rw_file_perms;
+')
Index: refpolicy/policy/modules/kernel/kernel.te
===================================================================
--- refpolicy.orig/policy/modules/kernel/kernel.te
+++ refpolicy/policy/modules/kernel/kernel.te
@@ -153,6 +153,9 @@ genfscon proc /sys/net/unix gen_context(
 type sysctl_vm_t, sysctl_type;
 genfscon proc /sys/vm gen_context(system_u:object_r:sysctl_vm_t,s0)
 
+type sysctl_vm_overcommit_t, sysctl_type;
+genfscon proc /sys/vm/overcommit_memory gen_context(system_u:object_r:sysctl_vm_overcommit_t,s0)
+
 # /proc/sys/dev directory and files
 type sysctl_dev_t, sysctl_type;
 genfscon proc /sys/dev gen_context(system_u:object_r:sysctl_dev_t,s0)
Index: refpolicy/policy/modules/system/logging.te
===================================================================
--- refpolicy.orig/policy/modules/system/logging.te
+++ refpolicy/policy/modules/system/logging.te
@@ -447,6 +447,8 @@ kernel_read_kernel_sysctls(syslogd_t)
 kernel_read_proc_symlinks(syslogd_t)
 # Allow access to /proc/kmsg for syslog-ng
 kernel_read_messages(syslogd_t)
+# rsyslog
+kernel_read_vm_overcommit_sysctl(syslogd_t)
 kernel_read_vm_sysctls(syslogd_t)
 kernel_clear_ring_buffer(syslogd_t)
 kernel_change_ring_buffer_level(syslogd_t)
